bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java

```diff
@@ -475,7 +475,7 @@ public void thingUpdated(final Thing thing, ThingTrackerEvent thingTrackerEvent)
 
     private ThingHandler replaceThing(Thing oldThing, Thing newThing) {
         final ThingHandler thingHandler = thingHandlers.get(newThing.getUID());
-        if (oldThing != newThing) {
+        if (oldThing.equals(newThing)) {
```

Comment:
Nope, that was by intention!

We currently don't create copies and equals only goes against the UID - which by definition won't change.